src/mempool/mempool.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable } from "@nestjs/common";
+import { Cron, CronExpression } from "@nestjs/schedule";
+import { getRPCProvider } from "../utils/client";
+import { TransactionIdentifier } from "../models/TransactionIdentifer";
+import { OperationFactory } from "../models/OperationFactory";
+import { Transaction } from "../models/Transaction";
+
+@Injectable()
+export class MempoolService {
+  
+  private provider = getRPCProvider();
+  private mempool:TransactionIdentifier[] = [];  
+
+  constructor() {        
+
+    this.provider.on("pending", (tx) => {      
+      this.mempool.push(new TransactionIdentifier(tx.hash))      
+    })
+    
+  }
+
+  public async getAllTransactions() {
+    return {
+      "transaction_identifiers": this.mempool      
+    }
+  }
+
+  public async getTransaction(transactionHash: string) {            
+    
+    const transaction = await this.provider.getTransaction(transactionHash)
+    const operationFactory = new OperationFactory()
+
+    const transfer = operationFactory.transferEWT(transaction.from, transaction.to, transaction.value, true);
+    
+    return {
+      "tansaction": new Transaction(new TransactionIdentifier(transaction.hash), [
+      ...transfer])
+    }
+  }
+
+  @Cron(CronExpression.EVERY_5_SECONDS)
+  mempoolWatchCron() {
+
+    this.mempool.forEach(tx => {
+      this.provider.getTransaction(tx.hash).then(transaction => {
+        if (transaction && transaction.blockNumber && transaction.confirmations > 0){          
+          this.mempool.splice(this.mempool.findIndex(x => x.hash === transaction.hash) , 1)
+        }
+        
+      })      
+    })
+  }
```

Comment:
In general this this is a good approach, but I guess more optimal would be to:

1) using provider.on("block") to get block and all mined transactions
2) remove mined from the local cache

Reply:
Actually only using events is an unreliable method, specially on RPCs. As per my testing when only using block events there is a very high chance that you might not receive some of the events and then the item in the cache will never get removed.
Therefore a combination of block events and then a longer interval scheduler (i.e. 30 seconds) is the best approach to ensure cache always contains valid records.